File: src/main/java/br/com/inatel/sherlock/models/Product.java
package br.com.inatel.sherlock.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
public class Product {
    @Id
    private Long id;

    @JsonProperty("client_id")
    private Long clientId;

    @JsonProperty("drawer_id")
    private Long drawerId;

    private String name;

    private Double quantity;

    private Double weight;

    private String dimension;

    @JsonProperty("estimated_date")
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime estimatedDate;

    @JsonProperty("departure_date")
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime arrivalDate;

    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime departureDate;

    @JsonProperty("pass_code")
    private String passCode;

    private Boolean available;
}

File: backend/internal/model/product.go
package model

type Product struct {
	ID             int     `json:"id"`
	ClientID       int     `json:"client_id"`
	DrawerID       int     `json:"drawer_id"`
	DrawerLockerID int     `json:"drawer_locker_id"`
	Quantity       float64 `json:"quantity"`
	Weight         float64 `json:"weight"`
	Name           string  `json:"name"`
	Dimension      string  `json:"dimension"`
	EstimatedDate  string  `json:"estimated_date"`
	Address        string  `json:"address"`
	PassCode       string  `json:"pass_code"`
	Available      bool    `json:"available"`
}
